docs/proposals/migration/design-of-seamless-cluster-migration-scheme.md

```diff
@@ -0,0 +1,353 @@
+     // Workload represents the manifest workload to be deployed on managed cluster.
+     Workload WorkloadTemplate `json:"workload,omitempty"`
+ }
+ 
+ type WorkloadTemplate struct {
+     Manifests []Manifest `json:"manifests,omitempty"`
+ }
+ 
+ type Manifest struct {
+     runtime.RawExtension `json:",inline"`
+ }
+```
+
+
+
+### Process Logic
+
+1、**ResourceDetector：**
+
+Change: Assign the `conflictResolution` value of `PropagationPolicy` to `ResourceBinding` 
+
+
+
+2、**ResourceBinding Controller：**
+
+Change 1: Assign the `conflictResolution` of `ResourceBinding` to `Work`
+
+Change 2: Update the value of the `conflict-resolution` annotation in `workload`  based on the `conflictResolution` field and the original `conflict-resolution` annotation in `workload`
+
+
+
+3、**Execution Controller：**
+
+> Previous：determine whether the `work.karmada.io/conflict-resolution` annotation is included in the `workload` of `Work` . Only with this annotation and a value of  `overwrite` will the `workload` be synchronized to member clusters
+
+no changes
+
+
+
+### Test Plan
+
+Add e2e test cases:
+
+1）Create a host cluster and member clusters, install `Karmada` in the host cluster, and joins the member clusters.
+
+2）Creating a `Deployment` in a member cluster.
+
+3）Create a `PropagationPolicy` and `ResourceTemplate` in the host cluster, and verify whether the takeover results of `Deployments` that already exist in the member cluster match the expected values for the `conflictResolution` and other fields mentioned above.
+
+<br/>
+
+## Alternatives
+
+
+
+### Abandon old annotation method
+
+**Q：Should the old annotation method be abandoned ?**
+
+A：The old annotation method does have fewer usage scenarios, users might have a higher probability of using new methods, but I prefer to reserve the old annotation method, since:
+
+* compatible
+* Prevent individual special situations against `conflictResolution` value of `PropagationPolicy`, give user more flexibility to define configuration.
+
+
+
+### Less API modify
+
+**Q:  Whether is needed to modify the api of `ResourceBinding` and `Work` ?** 
+
+A：Honestly, there is no need to modify the api of `ResourceBinding` and `Work` from the implementation perspective (`conflictResolution` can be declared through annotations either). No such api modify even makes code more clean, but two reasons are under my consideration:
+
+1）We not only want to just meet the scenario in above stories, but also solemnly recommend the ability and practice of seamless migration to users. Adding this field to CRDs including `ResourceBinding` and `Work ` can more clearly demonstrate this ability to users than adding annotations.
+
+2）Adding annotations is just a **compatible** way for individual exceptions, even if we remove it, it's still justifiable. Assuming it doesn't exist, we still need to modify the api of `ResourceBinding` and `Work`. I mean, the annotation is just a addons, our desgin shouldn't overdependence on it.
+
```

Comment:
Will you provide a cmd to promote a cluster or clusters at one time, like "karmadactl promote -A -n default -c member1"?

Reply:
It's great idea!

Reply:
great idea, but I prefer to follow up this capability with other issues~